aspnetcore/host-and-deploy/iis/troubleshoot.md

```diff
@@ -15,46 +15,140 @@ uid: host-and-deploy/iis/troubleshoot
+The [ASP.NET Core Module](xref:fundamentals/servers/aspnet-core-module) attempts to start the worker process but it fails to start. The cause of a process startup failure can usually be determined from entries in the [Application Event Log](#application-event-log) and the [ASP.NET Core Module stdout log](#aspnet-core-module-stdout-log).
 
-One quick way to determine if the app is working properly is to run the app directly on Kestrel. If the app was published as a [framework-dependent deployment](/dotnet/core/deploying/#framework-dependent-deployments-fdd), execute `dotnet <assembly_name>.dll` in the deployment folder, which is the IIS physical path to the app. If the app was published as a [self-contained deployment](/dotnet/core/deploying/#self-contained-deployments-scd), run the app's executable directly from a command prompt, `<assembly_name>.exe`, in the deployment folder. If Kestrel is listening on default port 5000, the app should be available at `http://localhost:5000/`. If the app responds normally at the Kestrel endpoint address, the problem is more likely related to the reverse proxy configuration and less likely within the app.
+The *502.5 Process Failure* error page is returned when a hosting or app misconfiguration causes the worker process to fail:
 
-One way to determine if the reverse proxy is working properly is to perform a simple static file request for a stylesheet, script, or image from the app's static files in *wwwroot* using [Static File Middleware](xref:fundamentals/static-files). If the app can serve static files but MVC Views and other endpoints are failing, the problem is less likely related to the reverse proxy configuration and more likely within the app (for example, MVC routing or 500 Internal Server Error).
+![Browser window showing the 502.5 Process Failure page](troubleshoot/_static/process-failure-page.png)
 
-When Kestrel starts normally behind IIS but the app won't run on the system after successfully running locally, an environment variable can be temporarily added to *web.config* to set the `ASPNETCORE_ENVIRONMENT` to `Development`. As long as the environment isn't overridden in app startup, setting the environment variable allows the [developer exception page](xref:fundamentals/error-handling) to appear when the app is run. Setting the environment variable for `ASPNETCORE_ENVIRONMENT` in this way is only recommended for staging/testing servers that aren't exposed to the Internet. Be sure to remove the environment variable from the *web.config* file when finished. For information on setting environment variables via *web.config*, see [environmentVariables child element of aspNetCore](xref:host-and-deploy/aspnet-core-module#setting-environment-variables).
+**500 Internal Server Error**  
```

Comment:
There's a 3rd class of errors that occur part way through a response. It's too late for the server to send a 500 so you get a connection reset error instead. This happens quite a bit when serializing complex objects in an MVC action.